flink-core/src/main/java/org/apache/flink/api/common/accumulators/AverageAccumulator.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.api.common.accumulators;
+
+/**
+ * An accumulator that get the average values.
+ * Input can be {@code long}, {@code integer}, {@code double} as the result is {@code double}.
+ */
+public class AverageAccumulator implements SimpleAccumulator <Double> {
+
+	private double localValue = 0;
+	private long count = 0;
+
+	@Override
+	public void add(Double value) {
+		this.count++;
+		this.localValue += value;
+	}
+
+	public void add(double value) {
+		this.count++;
+		this.localValue += value;
+	}
+
+	public void add(Long value) {
```

Comment:
I`m not quite sure.
Just the same as the class LongCounter .
You can see the code following:

public class LongCounter implements SimpleAccumulator<Long> {

	private static final long serialVersionUID = 1L;
	
	private long localValue = 0;

	/**
	 * Consider using {@link #add(long)} instead for primitive long values
	 */
	@Override
	public void add(Long value) {
		this.localValue += value;
	}

	public void add(long value){
		this.localValue += value;
	}

	@Override
	public Long getLocalValue() {
		return this.localValue;
	}
	
	@Override
	public void merge(Accumulator<Long, Long> other) {
		this.localValue += other.getLocalValue();
	}
	
	@Override
	public void resetLocal() {
		this.localValue = 0;
	}

	@Override
	public LongCounter clone() {
		LongCounter result = new LongCounter();
		result.localValue = localValue;
		return result;
	}

	@Override
	public String toString() {
		return "LongCounter " + this.localValue;
	}
}